src/macros.rs

```diff
@@ -720,3 +720,55 @@ macro_rules! const_unreachable {
         loop {}
     }};
 }
+
+/// Asserts at compile time that `$condition` is true for `Self` or the given
+/// `$tyvar`s. Unlike `const_assert`, this is *strictly* a compile-time check.
+/// The condition is checked after monomorphization and, upon failure, emits a
+/// compile error.
```

Comment:
`const_assert!` is also strictly a compile-time check so long as it's always called from `const` code; its fanciness has to do with its behavior on older toolchains that don't support explicitly panicking in `const` code. Maybe I should update the docs to be more clear about that.

The distinction is actually that `static_assert` uses post-monomorphization errors only, and so its errors are generated during a different compiler phase. Arguably we could just inline `static_assert` into `ct_assert_dst_is_not_zst` below.